plugins/uefi-dbx/fu-uefi-dbx-snapd-notifier.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2024 Maciej Borzecki <maciej.borzecki@canonical.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+#include "config.h"
+
+#include <curl/curl.h>
+#include <curl/easy.h>
+
+#include "fu-uefi-dbx-snapd-notifier.h"
+
+struct _FuUefiDbxSnapdNotifier {
+	GObject parent_instance;
+	CURL *curl_template;
+	struct curl_slist *req_hdrs;
+};
+
+G_DEFINE_TYPE(FuUefiDbxSnapdNotifier, fu_uefi_dbx_snapd_notifier, G_TYPE_OBJECT)
+
+FuUefiDbxSnapdNotifier *
+fu_uefi_dbx_snapd_notifier_new(void)
+{
+	return g_object_new(FU_TYPE_UEFI_DBX_SNAPD_NOTIFIER, NULL);
+}
+
+static void
+fu_uefi_dbx_snapd_notifier_init(FuUefiDbxSnapdNotifier *self)
+{
+	self->curl_template = curl_easy_init();
+
+	/* TODO support system wide snapd socket for outside of snap scenarios */
+	if (!fu_snap_is_in_snap()) {
+		g_warning("attempted use of snapd notifier outside of snap");
```

Comment:
This will happen if compiled with snapd support which I think will be relatively common since it's a meson tri state feature gated upon curl and Linux only.

So I think you should downgrade this to "info" level (it won't print then) and return here.